hydroflow_lang/src/graph/partitioned_graph.rs

```diff
@@ -168,30 +170,44 @@ impl PartitionedGraph {
                             let iter_code =
                                 (op_constraints.write_iterator_fn)(&context_args, &iter_args);
 
-                            subgraph_op_iter_code.push(quote! {
-                                let #ident = #iter_code;
-                            });
+                            subgraph_op_iter_code.push(iter_code);
                         }
                     }
 
                     {
                         let pull_to_push_idx = pull_to_push_idx;
                         let pull_ident =
                             self.node_id_as_ident(subgraph_nodes[pull_to_push_idx - 1], false);
+
                         let push_ident =
-                            self.node_id_as_ident(subgraph_nodes[pull_to_push_idx], false);
+                            if let Some(&node_id) = subgraph_nodes.get(pull_to_push_idx) {
+                                self.node_id_as_ident(node_id, false)
+                            } else {
+                                // Entire subgraph is pull.
+                                assert_eq!(
```

Comment:
Is there a test that triggers this case?

Reply:
Yeah a few. Seems to hit this case pretty often when you're pulling from one or more sources and pushing to a single handoff (the handoff "push" doesn't count when I say "Entire subgraph is pull.")